### ISSUE TITLE
Attempting to click modify in the UI on a Server Group from inside Pipeline Configuration gets a 404

### ISSUE THREAD

**Issue by ryan-mf** [spinnaker/spinnaker#2025]
*Title*

Attempting to click modify in the UI on a Server Group from inside Pipeline Configuration gets a 404

*Cloud Provider*

GCP and Kubernetes

*Environment*

We use Spinnaker to trigger off of newly available container image tags in our GCP registry, and deploy those new tags to dev/stage/prod in our Kubernetes (Google Container Engine) clusters.

*Feature Area*

UI and Pipelines

*Description*

Navigating in the UI to the settings for an existing pipeline, going to the "Deploy" step, and clicking the "Edit" button for a given Server Group (Kube Deployment), results in nothing happening, so you can't edit the settings. If you inspect errors in Chrome, you see the following 404 error:
```
Error: [$compile:tpload] Failed to load template: app/scripts/modules/kubernetes/serverGroup/configure/wizard/wizard.html (HTTP status: 404 Not Found)
http://errors.angularjs.org/1.6.4/$compile/tpload?p0=app%2Fscripts%2Fmodules%2Fkubernetes%2FserverGroup%2Fconfigure%2Fwizard%2Fwizard.html&p1=404&p2=Not%20Found
    at vendor.bundle.js:85344
    at handleError (vendor.bundle.js:105270)
    at processQueue (vendor.bundle.js:102110)
    at vendor.bundle.js:102154
    at Scope.$digest (vendor.bundle.js:103249)
    at Scope.$apply (vendor.bundle.js:103547)
    at done (vendor.bundle.js:97665)
    at completeRequest (vendor.bundle.js:97891)
    at XMLHttpRequest.requestLoaded (vendor.bundle.js:97819) "Possibly unhandled rejection: {}"
```

*Steps to Reproduce*

Have an existing pipeline that deploys to Kubernetes. Attempt to edit the Server Group in the "deploy" stage in your pipeline. See that the Deployment (Server Group) settings window no longer appears.

*Additional Details*

This behavior appears to have started after upgrading Spinnaker components on 10/6:
```
rwhite@spinnaker-test2:~$ grep -C 5 -i spinnaker /var/log/apt/history.log
<snip>
Start-Date: 2017-10-06  17:18:06
Commandline: apt-get upgrade --with-new-pkgs
Upgrade: spinnaker-gate:amd64 (4.6.3, 4.7.0), stackdriver-agent:amd64 (5.5.2-368.trusty, 5.5.2-369.trusty), spinnaker-deck:amd64 (2.1139.0, 2.1142.0), spinnaker-clouddriver:amd64 (1.683.0, 1.684.0), spinnaker-orca:amd64 (3.21.10, 3.22.3), spinnaker-halyard:amd64 (0.34.0-21, 0.35.0-22)
End-Date: 2017-10-06  17:18:20
```

I attempted to upgrade to the latest versions available again today and the issue is still present:
```
Start-Date: 2017-10-10  23:50:07
Commandline: apt-get upgrade --with-new-pkgs
Install: linux-image-4.4.0-97-generic:amd64 (4.4.0-97.120~14.04.1, automatic), linux-image-extra-4.4.0-97-generic:amd64 (4.4.0-97.120~14.04.1, automatic)
Upgrade: initramfs-tools-bin:amd64 (0.103ubuntu4.7, 0.103ubuntu4.8), spinnaker-clouddriver:amd64 (1.684.0, 1.685.0), spinnaker-orca:amd64 (3.22.3, 3.23.1), curl:amd64 (7.35.0-1ubuntu2.10, 7.35.0-1ubuntu2.11), initramfs-tools:amd64 (0.103ubuntu4.7, 0.103ubuntu4.8), libcurl3:amd64 (7.35.0-1ubuntu2.10, 7.35.0-1ubuntu2.11), linux-image-generic-lts-xenial:amd64 (4.4.0.96.80, 4.4.0.97.81), libcurl3-gnutls:amd64 (7.35.0-1ubuntu2.10, 7.35.0-1ubuntu2.11)
End-Date: 2017-10-10  23:54:10
```

**Comment by lwander** [comment]
I'm not seeing this error - maybe it's been fixed since the latest deck was released? 

My steps where:

1. Create a new application
2. Create a new pipeline
3. Add a deploy stage & configure a k8s replica set
4. Save the pipeline
5. Click edit, make changes
6. Save the pipeline

**Comment by ryan-mf** [comment]
Ok, I'll try upgrading to latest deck and see if that fixex it:
```
root@spinnaker-test2:~# apt-get update && apt-get upgrade --with-new-pkgs && apt-get autoremove
<snip>
The following packages will be upgraded:
  spinnaker-deck spinnaker-front50
2 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by ryan-mf** [comment]
Unfortunately I'm still seeing the issue:
![image](https://user-images.githubusercontent.com/25380659/31458563-4c750a2c-ae75-11e7-940d-1c1444673af4.png)

**Comment by lwander** [comment]
Can you point out your deck version?

**Comment by ryan-mf** [comment]
```rwhite@spinnaker-test2:~$ dpkg -s spinnaker-deck
Package: spinnaker-deck
Status: install ok installed
Priority: optional
Section: java
Installed-Size: 43324
Maintainer: travis
Architecture: all
Source: spinnaker-deck
Version: 2.1143.0
Depends: apache2 (>= 2.4.7)
Description: spinnaker-deck
 deck-ui```

**Comment by lwander** [comment]
Can you try a hard browser refresh?

**Comment by ryan-mf** [comment]
Tried that and both Chrome and Firefox, same issue. However, I did just notice (no idea when this started) that I'm actually logged in as anonymous, even though I haven't changed my auth config, and I've had Google Oauth configured and working for quite some time. So maybe that has something to do with it? I have no option to login, either, and I seem to be able to do anything else with pipelines without authenticating...

**Comment by lwander** [comment]
Ok cool, just making sure we've covered all our bases - let me play with some settings and try to reproduce this

**Comment by ryan-mf** [comment]
Tried running "/opt/spinnaker/bin/reconfigure_spinnaker.sh" but no dice, still just anonymous access is allowed. I haven't moved to Halyard yet, which by coincidence I started working on today (planning to run Halyard from a container and deploy Spinnaker to Kubernetes). Maybe there was a code change that made the previous instructions for enabling Oauth with Google, invalid?

**Comment by lwander** [comment]
Yeah there was a gate change some time ago that required a config change - just for the sake of clarity could you file a separate issue that we can post the change to? Will make it easier to find in the future too.

**Comment by lwander** [comment]
But I don't think that these are related issues

**Comment by lwander** [comment]
A couple more things to try

1. Can you add another k8s server group in that pipeline stage?
2. Does the ad-hoc deploy screen work?
3. Have you maybe enabled the v2 provider and are using it against v1 pipelines?

**Comment by ryan-mf** [comment]
For #3, I def haven't started using the v2 provider, unless I'm somehow using it unknowingly. My GKE config hasn't changed in months.

I'll look at #'s 1 & 2.

**Comment by lwander** [comment]
Hmm maybe one more thing, can you check your settings.js to see if `features.versionedProviders` is enabled?

**Comment by ryan-mf** [comment]
`root@spinnaker-test2:/opt/spinnaker# grep -Ri features.versionedProviders *`

No results.

**Comment by lwander** [comment]
Ah maybe that's it, can you enable it?

**Comment by ryan-mf** [comment]
Sure, but I'm not sure which fils(s) to enable in, and under which header:
/opt/deck/html/settings.js, /opt/spinnaker/config/settings.js, or both?

**Comment by lwander** [comment]
It needs to end up in /opt/deck/html/settings.js. But I don't know how you're managing that file or what path you've taken to manage your other config.

**Comment by ryan-mf** [comment]
I haven't moved to Halyard yet, and have only directly edited /opt/spinnaker/config/<daemon>-local.yml (spinnaker, gate, clouddriver).

So, I'll assume I just need to directly edit `/opt/deck/html/settings.js`

**Comment by lwander** [comment]
:+1:

**Comment by ryan-mf** [comment]
Unfortunately that didn't do it. I'm going to continue working on moving Spinnaker to Halyard + Kube and see if it works there. Since I store my pipelines and such in a GCP storage bucket, once I can get Spinnaker to start, run, and authenticate properly in GKE, I'll just stop the current Spinnaker instance and remap the Kube version to that bucket, and hopefully it'll load up my pipelines without issue. From there I'll test again.

**Comment by lwander** [comment]
Yeah I've helped people migrate & that pattern worked w/o issue. It's important to keep your account names identical.

Also that auth issue you saw is handled by Halyard

**Comment by lwander** [comment]
Seeing that this is fixed & we're aiming to release 1.5.0 in place of 1.4.3 I will close this.
